src/coins.cpp

```diff
@@ -941,25 +942,27 @@ bool CCoinsViewCache::IsCertApplicableToState(const CScCertificate& cert, int nH
 
 
     // Retrieve previous end epoch block info for certificate proof verification
-    int prev_start_epoch_block_height = scInfo.StartHeightForEpoch(cert.epochNumber-1);
-    int prev_end_epoch_block_height   = scInfo.StartHeightForEpoch(cert.epochNumber) - 1;
+    int prev_end_epoch_block_height = scInfo.StartHeightForEpoch(cert.epochNumber) - 1;
+    int curr_end_epoch_block_height = scInfo.StartHeightForEpoch(cert.epochNumber + 1) - 1;
```

Comment:
Minor improvement. We may introduce `EndHeightForEpoch` method to avoid every time doing like `scInfo.StartHeightForEpoch(cert.epochNumber + 1) - 1)`
So, instead here and in a lot of other places we may type:
`scInfo.EndHeightForEpoch(cert.epochNumber)`
Seems much more clear to me.

`EndHeightForEpoch` may reuse the logic of `StartHeightForEpoch`
What do you think @alsala @VolodymyrBabaiev ?

Reply:
Ok for me, it is useful

Reply:
Yes, make sense.